src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -38,8 +43,86 @@ export const ADMIN_COOKIE_OPTIONS_WITH_EXPIRY = {
 
 const logger = createLoggerWithLabel(module)
 
-const serveFormReact: ControllerHandler = (_req, res) => {
-  const reactFrontendPath = path.resolve('dist/frontend')
+const reactFrontendPath = path.resolve('dist/frontend')
+const reactHtml = readFileSync(path.join(reactFrontendPath, 'index.html'), {
+  encoding: 'utf8',
+})
+
+type MetaTags = {
+  title: string
+  description: string
+  image: string
+}
+const replaceWithMetaTags = ({
+  title,
+  description,
+  image,
+}: MetaTags): string => {
+  return (
+    reactHtml
+      // there are multiple __OG_TITLE__ and we are unable to use string.replaceAll so we use regexp
+      .replace(new RegExp('__OG_TITLE__', 'g'), title)
+      .replace('<title>FormSG</title>', `<title>${title}</title>`)
+      .replace('__OG_DESCRIPTION__', description)
+      .replace('__OG_IMAGE__', image)
```

Comment:
Some comments:
* no need for the comment 😅 
* don't use the `RegExp()` constructor unless you are constructing the regex pattern from variables, for static patterns, use the slash-slash syntax instead
* ⚠️ there are also multiple `__OG_DESCRIPTION__` and `__OG_IMAGE__` in the source file, all regexps should use the `global` flag on for all replacements
* Since we are using the same title value for the page main `<title>` tag, might as well replace the source `<title>FormSG</title>` into `<title>__OG_TITLE__</title>`, and remove the title-specific replacement statement

```suggestion
      .replace(/__OG_TITLE__/g, title)
      .replace(/__OG_DESCRIPTION__/g, description)
      .replace(/__OG_IMAGE__/g, image)
```

Also just noting: this is doing html injection in text nodes and attributes. We know from the values below that the entries are static and html-safe (thankfully! - meaning no markup of xss injection possible), but we should consider doing html-encoding for safety anyway in this function.

Ref: [OWASP recommendations on html and attribute encoding](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#RULE_.231_-_HTML_Escape_Before_Inserting_Untrusted_Data_into_HTML_Element_Content)